lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -234,8 +235,30 @@ static SmallVector<unsigned> eraseOrder(ArrayRef<unsigned> order,
   return resOrder;
 }
 
+SmallVector<unsigned> getOrderForDotOperand(unsigned opIdx, unsigned rank,
+                                            bool kMajor) {
+  // kMajor: if true, the matrix is fastest-running on k,
+  //         otherwise it is on m (resp. n)
+  // opIdx=0: [batch, m, k] if rank == 3 else [m, k]
+  // opIdx=1: [batch, k, n] if rank == 3 else [k, n]
+  // batch (if rank == 3) is always the slowest running dimension
+  assert(rank == 2 || rank == 3);
+  assert(opIdx == 0 || opIdx == 1);
+  SmallVector<unsigned> order(rank);
+  std::iota(order.rbegin(), order.rend(), 0);
+  // If opIdx is 1 and kMajor is true, the order is [0, 1]
+  // (resp. [1, 2, 0] if rank == 3)
+  // Same if opIdx is 0 and kMajor is false
+  if (bool(opIdx) == kMajor) {
+    std::swap(order[0], order[1]);
+  }
+  return order;
+}
+
 SmallVector<unsigned> getWarpOrder(Attribute layout) {
   auto order = getOrder(layout);
+  // FIXME: This path should be removed, and we should just return
+  // getOrderForDotOperand(kMajor=false) for both mma and dotOperand
   if (auto mmaLayout = dyn_cast<NvidiaMmaEncodingAttr>(layout)) {
```

Comment:
The comment is a bit confusing for me. Why should we use `getOrderForDotOperand` for MMA?

Reply:
my point is that we should unify both paths and simply return `getOrderForDotOperand(opIdx=0)` for mma. But yes, the comment is confusing. Let me improve it.